zeppelin-server/src/main/java/org/apache/zeppelin/utils/SecurityUtils.java

```diff
@@ -88,13 +89,24 @@ public static String getPrincipal() {
 
     String principal;
     if (subject.isAuthenticated()) {
-      principal = subject.getPrincipal().toString();
+      principal = extractPrincipal(subject);
     } else {
       principal = ANONYMOUS;
     }
     return principal;
   }
 
+  private static String extractPrincipal(Subject subject) {
+    String principal;
+    Object principalObject = subject.getPrincipal();
+    if (principalObject instanceof Principal) {
+      principal = ((Principal) principalObject).getName();
+    } else {
+      principal = String.valueOf(principalObject);
```

Comment:
what's the reason for `String.valueOf` vs `.toString()` from before?

Reply:
That is because if the argument is null, then a string equal to "null" is returned instead of NullPointerException (just in case).